### PR TITLE
VisionTransformer Loss Compare

### PR DESCRIPTION
**Pull request by rentainhe**
## TODO
这个PR的目的是, 维护一个可以做vision transformer loss对齐的分支
- [ ] 添加torch vision transformer训练相关的coding
- [ ] 修改trainer和vit的cfg, 训练libai的vit模型，并保存Loss
- [ ] 添加画图函数

### CODE REVIEW

File: libai/data/samplers/compare_loss_sampler.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        epoch = self.consumed_samples // self.data_size_per_epoch
+        current_epoch_samples = self.consumed_samples % self.data_size_per_epoch
+        batch = []
+
+        while True:
+            bucket_offset = current_epoch_samples // self.data_parallel_size
+            start_idx = self.data_parallel_rank * self.data_size_per_epoch
+
+            if self.shuffle:
+                np.random.seed(0)
```
Comment (rentainhe) [inline libai/data/samplers/compare_loss_sampler.py:76]: 这个位置原本是`np.random.seed(self.seed)`, 但是发现不生效， 因此手动赋值`seed(0)`

File: train_torch_vit.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    batch_sampler=CyclicSampler(dataset=imagenet_dataset, micro_batch_size=BATCH_SIZE, shuffle=True),
+)
+
+
+"""
+Training and Write Loss
+"""
+model = vit_tiny_patch16_224()
+# 第一次执行需要保存一下权重, 之后可以把这个注释了只用一个权重
+# torch.save(model.state_dict(), "./torch_vit_tiny_weight.pth")
```
Comment (rentainhe) [inline train_torch_vit.py:140]: 第一次运行的时候需要把这个注释给关闭， 保存一个权重，之后再注释
